### ISSUE TITLE
🤗 IPv6 support

### ISSUE THREAD

**Issue by suntong** [gofiber/fiber#1154]
**Question description**

In #601, IPv6 support was added

https://github.com/gofiber/fiber/blob/b8fbaca1ea0b1a80c34257baa0a9c12e49c11de4/app.go#L111-L113

and tcp4 + tcp6 was default.

Now, 

https://github.com/gofiber/fiber/blob/3b956ef3a81de3228afb76bc5f56e1db5f797e39/app.go#L559

It seems only tcp4 is supported. 

Why is that, and how can I enable IPv6 again?

Thx

**Code snippet** _Optional_

```go
package main

import "github.com/gofiber/fiber/v2"

func main() {
  app := fiber.New()
  // ..
}
```

**Comment by kiyonlin** [comment]
The primary reason is because of [prefork](https://github.com/gofiber/fiber/blob/master/prefork.go#L38-L39), we can't use `tcp` in prefork mode. And in additional, most cases are using `tcp4`, so we removed `tcp6` and only used `tcp4` by default in v2.

If you indeed need to start a `tcp6` server, we can discuss here.

**Comment by suntong** [comment]
Ok, understand now. 

I do need to start a tcp6 server besides tcp4, what options I have?

**Comment by kiyonlin** [comment]
Made a PR(#1155), we can move the discussion there.

**Comment by suntong** [comment]
Now that the PR has been closed, I'll continue the discussion here...

> we can't use tcp in prefork mode.

I'd like to know more about that, because I can see that both tcp6 and tcp4 work just fine with prefork mode. So what was the problem?

The reason I'm asking is that, previously, besides _" "tcp4" (IPv4-only), "tcp6" (IPv6-only) "_, there is also a "`tcp`" setting that allows starting a tcp6 server _besides_ tcp4, which is the mode I was hoping to get...

**Comment by kiyonlin** [comment]
That PR hasn't been merged yet. 

`tcp` doesn't work for linux and darwin in prefork mode. But it may works for windows which I don't have test environment.

**Comment by suntong** [comment]
Oh, I don't care about Windows, 

Hmm, in that case, OK, I can live with only one, not both. 

thx for the PR, @kiyonlin

**Comment by suntong** [comment]
Hmm,

> WARNING: When prefork is set to true, only "tcp4" and "tcp6" can be chose.

I don't use prefork, can I use `NetworkTCP` to have both tcp4 & tcp6 server then?

**Comment by kiyonlin** [comment]
> Hmm,
> 
> > WARNING: When prefork is set to true, only "tcp4" and "tcp6" can be chose.
> 
> I don't use prefork, can I use `NetworkTCP` to have both tcp4 & tcp6 server then?

Of course

**Comment by suntong** [comment]
OK, waiting it merge to try `NetworkTCP` and my linux **_and_** darwin

**Comment by kiyonlin** [comment]
Released v2.5.0
